soroban-sdk/src/env.rs

```diff
@@ -576,6 +578,91 @@ impl Env {
         env
     }
 
+    /// Enables detailed metering of contract invocations.
+    ///
+    /// The top-level contract invocations and lifecycle operations (such as
+    /// `register` or `env.deployer()` operations) will be metered and the
+    /// budget will reset in-between them. Metering will not be reset while
+    /// inside the call (e.g. if a contract calls or creates another contract,
+    /// that won't reset metering).
+    ///
+    /// The metered resources for the last invocation can be retrieved with
+    /// `get_last_invocation_resources()` and the estimated fee corresponding
+    /// to these resources can be estimated with
+    /// `estimate_last_invocation_fee()`.
+    ///
+    /// While the resource metering may be useful for contract optimization,
+    /// keep in mind that resource and fee estimation may be imprecise. Use
+    /// simulation with RPC in order to get the exact resources for submitting
+    /// the transactions to the network.
+    pub fn enable_invocation_metering(&self) {
```

Comment:
What cost does having this enabled incur? Is it noticeable? If it's not material we should enable this by default.

Instead of a function here, I think we should move this into the EnvTestConfig. It'll keep the surface area of testutils functions lower, and is then easily discoverable there. But I'd still have it enabled by default.

Reply:
The cost is not material, but this does change the observed behavior as the budget will be reset in-between invocations. So in theory this might break someone's tests, which is why I was hesitant about enabling this by default (maybe that should have been enabled in the major release, but now we have to wait until the next one).

Reply:
> the budget will be reset in-between invocations

I think this is a good thing, we should I think have some consistency across these types of functions where top-level invocation is seen as a single unit, much like how you can get auth for the last invocation.

> maybe that should have been enabled in the major release

+1.

Reply:
Yeah, I think we're on the same page here, I'd like to enable this by default ASAP, but probably it's safer to wait until the next release. We can make an announcement though and add some examples.